templates/csharp/m365-message-extension/Search/SearchApp.cs.tpl

```diff
@@ -0,0 +1,77 @@
+{ 
+    // Search
+    protected override async Task<MessagingExtensionResponse> OnTeamsMessagingExtensionQueryAsync(ITurnContext<IInvokeActivity> turnContext, MessagingExtensionQuery query, CancellationToken cancellationToken)
+    {
+        var text = query?.Parameters?[0]?.Value as string ?? string.Empty;
+        var packages = await FindPackages(text);
+        // We take every row of the results and wrap them in cards wrapped in in MessagingExtensionAttachment objects.
+        // The Preview is optional, if it includes a Tap, that will trigger the OnTeamsMessagingExtensionSelectItemAsync event back on this bot.
+        var attachments = packages.Select(package =>
+        {
+            var previewCard = new ThumbnailCard { Title = package.Item1 };
+            var adaptiveCard = new AdaptiveCard(new AdaptiveSchemaVersion("1.4"))
+            {
+                Body = new List<AdaptiveElement>
+            {
+                new AdaptiveTextBlock
+                {
+                    Text = package.Item1,
+                    Size = AdaptiveTextSize.Large,
+                    Wrap = true
+                },
+                new AdaptiveTextBlock
+                {
+                    Text = package.Item3,
+                    Size = AdaptiveTextSize.Medium,
+                    Wrap = true
+                }
+            }
+            };
+            if (!string.IsNullOrEmpty(package.Item5))
+            {
+                previewCard.Images = new List<CardImage>() { new CardImage(package.Item5, "Icon") };
+                adaptiveCard.Body.Insert(0,new AdaptiveImage()
+                {
+                    Url = new Uri(package.Item5),
+                    Style = AdaptiveImageStyle.Person,
+                    Size = AdaptiveImageSize.Small,
+                });
+            }
+            var attachment = new MessagingExtensionAttachment
+            {
+                ContentType = AdaptiveCard.ContentType,
+                Content = adaptiveCard,
+                Preview = previewCard.ToAttachment()
+            };
+
+            return attachment;
+        }).ToList();
+
+        return new MessagingExtensionResponse
+        {
+            ComposeExtension = new MessagingExtensionResult
+            {
+                Type = "result",
+                AttachmentLayout = "list",
+                Attachments = attachments
+            }
+        };
+    }
+
+    // Generate a set of substrings to illustrate the idea of a set of results coming back from a query. 
+    private async Task<IEnumerable<(string, string, string, string, string)>> FindPackages(string text)
+    {
+        var obj = JObject.Parse(await (new HttpClient()).GetStringAsync($"https://azuresearch-usnc.nuget.org/query?q=id:{text}&prerelease=true"));
+        return obj["data"].Select(item => (item["id"].ToString(), item["version"].ToString(), item["description"].ToString(), item["projectUrl"]?.ToString(), item["iconUrl"]?.ToString()));
```

Comment:
This line seems unreadable. 
I know Teams Samples also implement it in this way.
Maybe we can just change some style, like make all the item[X].ToString() lines paralleled.